include/mbedtls/pk.h

```diff
@@ -199,7 +287,13 @@ void mbedtls_pk_free( mbedtls_pk_context *ctx );
  *                  MBEDTLS_ERR_PK_BAD_INPUT_DATA on invalid input,
  *                  MBEDTLS_ERR_PK_ALLOC_FAILED on allocation failure.
  *
- * \note            For contexts holding an RSA-alt key, use
+ * \note            Engines that implement of opaque keys may offer an
+ *                  alternative setup function that take engine-dependent
+ *                  parameters. If such a function exists, call it
```

Comment:
When should alternative implementations do the setup in their `pk_info`'s `calloc_func`, and when should they provide their own setup function?

Reply:
I'm not sure if I understand your question correctly, but if the setup requires anything beyond allocating and dealing with the context (and that's often the case), like initializing hardware - it's the responsibility of the setup function.